### PR TITLE
email/mu4e: add featurep check for workspaces

### PR DESCRIPTION
**Pull request by dsdshcym**
- if module workspaces is not used, an error would be raised when calling `=mu4e`
  `=mu4e: Symbol’s function definition is void: +workspace-switch`
- if module workspaces is not used, an error would be raised when quiting mu4e
  `=mu4e: Symbol’s function definition is void: +workspace/delete`

This PR fixes these two issues.

I think there might be a better fix:
Like extracting these workspaces tweeks to a new file and adding a `;;;###if (featurep! :ui workspaces)` check at the beginning?

### CODE REVIEW

**Comment by hlissner** [comment]
I've been meaning to generalize these two, but haven't gotten around to it. Rather than just skip over the `+workspace-switch` call, I'd rather it stow away the window config and `delete-other-windows` before opening mu4e, then restore the window config in `+mu4e-kill-mu4e-h`.

This is how the [twitter](https://github.com/hlissner/doom-emacs/blob/develop/modules/app/twitter/autoload.el#L32-L36) and [calendar](https://github.com/hlissner/doom-emacs/blob/develop/modules/app/calendar/autoload.el#L23) modules do it, and how I had intended for the mu4e module to work.

Can you make these changes? Otherwise, I'll look into it sometime in the next few weeks.

**Comment by dsdshcym** [comment]
@hlissner implemented like what twitter module does, please check :)

File: modules/email/mu4e/autoload/email.el [hunk 1]
```diff
@@ -52,7 +53,11 @@ default/fallback account."
   "Start email client."
   (interactive)
   (require 'mu4e)
-  (+workspace-switch +mu4e-workspace-name t)
+  (if (featurep :ui 'workspaces)
+      (+workspace-switch +mu4e-workspace-name t)
+    (setq +mu4e--old-wconf (current-window-configuration)))
+  (delete-other-windows)
+  (switch-to-buffer (doom-fallback-buffer))
```
Comment (hlissner) [inline modules/email/mu4e/autoload/email.el]: Please move these two lines into the if block (after setting `+mu4e--old-wconf`). `+workspace-switch` already does this for us and could be a problem later.

Comment (dsdshcym) [reply modules/email/mu4e/autoload/email.el]: updated in https://github.com/hlissner/doom-emacs/pull/2123/commits/f6a10f1268f72197c3e4e52b4f924e891907649d

**Comment by hlissner** [comment]
Looks good to me! Thanks again for your help!

**Comment by alexandergraul** [comment]
After updating to this commit, `=mu4e` does not rename the workspace to `*mu4e*` anymore. I've got `:ui workspaces` enabled, see included details.

<details>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND DBUS GSETTINGS GLIB NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS XWIDGETS LIBSYSTEMD
        build      Dez 08, 2019
        buildopts  (--disable-build-details --without-pop --with-mailutils --without-hesiod --with-gameuser=:games --with-kerberos --with-kerberos5 --with-file-notification=inotify --with-modules --enable-autodepend --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --localstatedir=/var --sharedstatedir=/var/lib --libexecdir=/usr/lib --enable-locallisppath=/usr/share/emacs/26.3/site-lisp:/usr/share/emacs/site-lisp --with-x --with-xim --with-sound --with-xpm --with-jpeg --with-tiff --with-gif --with-png --with-rsvg --with-dbus --with-xft --without-gpm --with-x-toolkit=gtk3 --with-toolkit-scroll-bars --x-includes=/usr/include --x-libraries=/usr/lib64 --with-libotf --with-m17n-flt --with-xwidgets --build=x86_64-suse-linux 'CFLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -D_GNU_SOURCE -DGDK_DISABLE_DEPRECATION_WARNINGS -DGLIB_DISABLE_DEPRECATION_WARNINGS -pipe -Wno-pointer-sign -Wno-unused-variable -Wno-unused-label -fno-optimize-sibling-calls -fno-PIE -DSYSTEM_PURESIZE_EXTRA=55000 	 -DSITELOAD_PURESIZE_EXTRA=10000 ' LDFLAGS=-Wl,-O2)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop, upstream/develop, upstream/HEAD 385ff35ad 2019-12-08 01:02:01 -0500
system  type       gnu/linux
        config     x86_64-suse-linux-gnu
        shell      /usr/bin/zsh
        uname      Linux 5.3.12-1-default #1 SMP Thu Nov 21 07:21:43 UTC 2019 (a6f6081) x86_64
        path       (~/.rbenv/shims/ ~/go/bin/ ~/bin/ ~/.cargo/bin/ ~/.yarn/bin/ ~/.rbenv/bin/ ~/.cabal/bin/ ~/.emacs.d/bin/ ~/.local/bin/ ~/bin/ /usr/local/bin/ /usr/bin/ /bin/ /usr/games/ ~/.fzf/bin/ /usr/lib/emacs/26.3/x86_64-suse-linux/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company (ivy +icons) :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +all +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold format multiple-cursors rotate-text snippets word-wrap :emacs (dired +icons) electric vc :term vterm :tools (eval +overlay) flycheck flyspell (lookup +docsets) lsp magit :lang data emacs-lisp go (javascript +lsp) latex markdown (org +pandoc) plantuml (ruby +rbenv) sh web :email mu4e :app write :config literate (default +bindings +smartparens))
        packages   (n/a)
        elpa       (n/a)
```
</details>

**Comment by alexandergraul** [comment]
I've found the problem, it should be `(featurep! :ui workspaces)` instead of `(featurep :ui 'workspaces)`. I'll open a PR.

**Pull request merged**